pkg/process/build/build.go

```diff
@@ -64,38 +64,17 @@ func Build(ctx context.Context, pullRegistry docker.Registry, pushRegistry docke
 	auroraVersion := runtime.NewAuroraVersionFromBuilderAndBase(appVersion, snapshot,
 		application.MavenGav.Version, buildImage, baseImage.DockerImage, deliverable.SHA1)
 
-	dockerBuildConfig, err := prepper(cfg, auroraVersion, deliverable, baseImage)
+	dockerBuildConfigArray, err := prepper(cfg, auroraVersion, deliverable, baseImage)
 	if err != nil {
 		return errors.Wrap(err, "Error preparing image")
 	}
 
-	tagWith := cfg.DockerSpec.TagWith
-	for _, buildConfig := range dockerBuildConfig {
-		if !buildConfig.AuroraVersion.Snapshot {
-			tags, err := pushRegistry.GetTags(ctx, cfg.DockerSpec.OutputRepository)
-			if err != nil {
-				return err
-			}
-			semanticVersion := buildConfig.AuroraVersion.GetGivenVersion()
-			completeVersion := buildConfig.AuroraVersion.GetCompleteVersion()
-			logrus.Debugf("GivenVersion=%s, CompleteVersion=%s", semanticVersion, completeVersion)
-			for _, tag := range tags.Tags {
-				repoTag := docker.ConvertTagToRepositoryTag(tag)
-				logrus.Debug(repoTag)
-				if strings.EqualFold(repoTag, completeVersion) {
-					return errors.Errorf("There are already a build with tag %s, overwrite not allowed", completeVersion)
-				}
-				if strings.EqualFold(repoTag, semanticVersion) {
-					return errors.Errorf("There are already a build with tag %s, overwrite not allowed", semanticVersion)
-				}
-				if strings.EqualFold(repoTag, tagWith) {
-					return errors.Errorf("Given value for TagWith=%s have already been build, overwrite not allowed", tagWith)
-				}
-			}
-		}
+	overrideError := checkAllTagsForOverride(ctx, dockerBuildConfigArray, pushRegistry, cfg)
```

Comment:
Synes den godt kan hete err, da skjønner man hva slags type det er

Reply:
Hadde en mulighet vært å sjekke først for snapshot? Om det ikke er snapshot, så kan man etterpå sjekke at tag ikke finnes fra før?

Reply:
1. overrideError -> err 
2. nei, det blir litt vanskelig da snapshot ligger inne i listen som sjekkes for TAG.